server/src/main/java/com/google/tsunami/security/scanner/utilities/TsunamiTestbedUtil.java

```diff
@@ -0,0 +1,159 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.tsunami.security.scanner.utilities;
+
+import com.google.common.collect.ImmutableList;
+import freemarker.template.TemplateException;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The internal implementation of grpc requests.
+ */
+final class TsunamiTestbedUtil {
+
+  private final Object lock;
+
+  TsunamiTestbedUtil() {
+    lock = new Object();
+  }
+
+  public String createDeployment(
+      String application, String configPath, String templateData, String deployerJobPath)
+      throws IOException, TemplateException, ApiException, InterruptedException {
+    synchronized (lock) {
+      String jobId = "initialID";
+
+      System.out.println(
+          "app: "
+              + application
+              + " config path: "
+              + configPath
+              + " template data: "
+              + templateData
+              + " job yaml path: "
+              + deployerJobPath);
+
+      BatchV1Api batchV1Api = new BatchV1Api();
+      CoreV1Api coreV1Api = new CoreV1Api();
+
+      File configFile = new File(deployerJobPath);
+      // Replace template data in deployer yaml file.
+      ImmutableMap<String, String> templateDataMap =
+          ImmutableMap.of(
+              "app", application, "configPath", configPath, "templateData", templateData);
+      String resourceConfig = FreeMarkerUtil.replaceTemplates(templateDataMap, configFile);
+
+      // Load deployer job yaml file.
+      Yaml.addModelMap("v1", "Job", V1Job.class);
+      V1Job v1Job = (V1Job) Yaml.load(resourceConfig);
+      System.out.println("Yaml Loaded.");
+
+      // Create deployer job.
+      batchV1Api.createNamespacedJob("default", v1Job, null, null, null);
+      System.out.println("Created Job");
+
+      // Wait for service to start.
+      while (!listApplications().contains(application)) {
+        System.out.println("Waiting for service to start...");
+        Thread.sleep(1000);
+      }
+
+      // Get Unique Id of certain application.
+      V1ServiceList v1ServiceList =
+          coreV1Api.listNamespacedService(
+              "default", null, null, null, null, null, null, null, null, null);
+
+      for (V1Service svc : v1ServiceList.getItems()) {
+        if (svc.getMetadata().getName().equals(application)) {
+          jobId = svc.getMetadata().getUid();
+          System.out.println("Application " + application + "'s unique id is: " + jobId);
+        }
+      }
+
+      return jobId;
+    }
+  }
+
+  public List<String> listApplications() throws ApiException {
```

Comment:
Return `ImmutableList` here to be more explicit that the returned value cannot be modified by the caller.